vyper/semantics/analysis/utils.py

```diff
@@ -416,7 +416,7 @@ def _is_empty_list(node):
 
 def _is_type_in_list(obj, types_list):
     # check if a type object is in a list of types
-    return any(i.compare_type(obj) for i in types_list)
+    return any(obj.compare_type(i) for i in types_list)
```

Comment:
Haven't figured out why the current form would throw for this valid contract, but compile once it is flipped:
```
@external
def foo() -> String[33]:
    for x in ["hello", ",", "world"]:
        if x == ",":
            return x
    return ""
```